contracts/ProofMarketplace.sol

```diff
@@ -742,42 +686,74 @@ contract ProofMarketplace is
+        return listOfBid.length;
+    }
 
     // TODO: add this function back(commented due to size)
     function marketCounter() external view returns (uint256) {
         return marketData.length;
     }
 
-    function setOperatorRewardShare(address _operator, uint256 _rewardShare) external {
-        operatorRewardShares[_operator] = _rewardShare;
-        emit OperatorRewardShareSet(_operator, _rewardShare);
+    /// @notice Set the reward share for a prover
+    // TODO: check if this can frontrun bid
+    function setProverRewardShare(uint256 _rewardShare) external {
+        proverRewardShares[_msgSender()] = _rewardShare;
+        emit ProverRewardShareSet(_msgSender(), _rewardShare);
     }
 
-    function _distributeOperatorFeeReward(address _operator, uint256 _feePaid) internal returns (uint256 feeRewardRemaining) {
-        // calculate operator fee reward
-        uint256 operatorFeeReward = Math.mulDiv(_feePaid, operatorRewardShares[_operator], 1e18);
-        feeRewardRemaining = _feePaid - operatorFeeReward;
+    function _distributeProverFeeReward(address _prover, uint256 _feePaid) internal returns (uint256 feeRewardRemaining) {
+        // calculate prover fee reward
+        uint256 proverFeeReward = Math.mulDiv(_feePaid, proverRewardShares[_prover], 1e18);
+        feeRewardRemaining = _feePaid - proverFeeReward;
 
-        // update operator fee reward
-        _increaseClaimableAmount(_operator, operatorFeeReward);
+        // update prover fee reward
+        proverClaimableFeeReward[_prover] += proverFeeReward;
 
-        emit OperatorFeeRewardAdded(_operator, operatorFeeReward);
+        emit ProverFeeRewardAdded(_prover, proverFeeReward);
     }
 
-    /// @dev updated by SymbioticStaking contract when job is completed
+    /// @notice Called when SymbioticStaking reward distributes fee rewards
     function distributeTransmitterFeeReward(address _transmitter, uint256 _feeRewardAmount) external onlyRole(SYMBIOTIC_STAKING_ROLE) {
-        _increaseClaimableAmount(_transmitter, _feeRewardAmount);
+        transmitterClaimableFeeReward[_transmitter] += _feeRewardAmount;
         emit TransmitterFeeRewardAdded(_transmitter, _feeRewardAmount);
     }
 
-    /// @dev Only SymbioticStakingReward can call this function
     function transferFeeToken(address _recipient, uint256 _amount) external onlyRole(SYMBIOTIC_STAKING_REWARD_ROLE) {
         IERC20(PAYMENT_TOKEN).safeTransfer(_recipient, _amount);
     }
 
+    //-------------------------------- Overrides start --------------------------------//
+    function supportsInterface(
+        bytes4 interfaceId
+    ) public view virtual override(ERC165Upgradeable, AccessControlUpgradeable) returns (bool) {
+        return super.supportsInterface(interfaceId);
+    }
+
+    /**
+     @notice Enforces PMP to use only one matching engine image
+     */
+    function setMatchingEngineImage(bytes calldata pcrs) external onlyRole(UPDATER_ROLE) {
```

Comment:
`setMatchingEngineImage` and `verifyMatchingEngine` functions aren't really overrides instead they are matching engine management functions and have to segregated.